### PR TITLE
User/chepel d/for merge

### PR DESCRIPTION
**Pull request by dmitrychepel**

### CODE REVIEW

File: Source/zcl_app.c [hunk 1]
```diff
@@ -215,156 +209,97 @@ static void zclApp_LedFeedback(void) {
...
-    }
-}
 static void zclApp_ReadSensors(void) {
-    if (zclApp_Config.LedFeedback) {
+  if (zclApp_Config.LedFeedback) {
         HalLedSet(HAL_LED_1, HAL_LED_MODE_BLINK);
     }
-    static uint8 currentSensorsReadingPhase = 0;
+    static uint8 currentSensorsReadingPhase = 0; 
+    static uint8 temp_sensor_type = 0; // 0 bme280, 1 ds18b20, 2 not found
```
Comment (nurikk) [inline Source/zcl_app.c:226]: Can we use something like enum or macros to define me types?

File: Source/zcl_app.c [hunk 2]
```diff
@@ -215,156 +209,97 @@ static void zclApp_LedFeedback(void) {
...
+      osal_pwrmgr_task_state(zclApp_TaskID, PWRMGR_HOLD);
+      (*air_dev->RequestMeasure)();
+      break;
+    case 1:     
+      ppm = (*air_dev->Read)();
+      if (ppm == AIR_QUALITY_INVALID_RESPONSE)
+      {
+        air_dev =  (air_dev == &sense_air_dev) ? &MHZ19_dev : &sense_air_dev;
+        LREPMaster("Sensor type UNKNOWN continue detect\r\n");
+        current_co2_sensor_state = 0xff; //a part of algorithm avoiding exceptional io
```
Comment (nurikk) [inline Source/zcl_app.c:244]: Kindly don't use magic numbers

File: Source/zcl_app.c [hunk 3]
```diff
@@ -215,156 +209,97 @@ static void zclApp_LedFeedback(void) {
...
+      {
+        zclApp_ReadBME280(&bme_dev);
+        break;
+      }
+      currentSensorsReadingPhase++;
+      //missed break means: do not initiate new read iteration in case of missing ds18b20 sensor
+    case 4:  
+      if (temp_sensor_type == 1) {
         temp = readTemperature();
         if (temp == 1) {
```
Comment (nurikk) [inline Source/zcl_app.c:274]: Same magic numbers

**Review by nurikk — CHANGES_REQUESTED** [review]

File: Source/zcl_app.c (hunk 3 above)
Comment (dmitrychepel) [reply Source/zcl_app.c:274]: fixed but "temp==1" leaved.
it was before, and I don't have DS18B20 to check it.
I see that readTemperature is buggy now.
"1" very bad constant for temperature device to indicate mistake of operation.

**Comment by nurikk** [comment]
Thanks

**Pull request merged**